test/rules/assertions/requireParam.js

```diff
@@ -84,6 +84,39 @@ export default {
           }
       `,
     },
+    {
+      code: `
+          /**
+           * @param foo
+           */
+          function quux (foo, bar, {baz}) {
+
+          }
+      `,
+      errors: [
+        {
+          message: 'Missing JSDoc @param "bar" declaration.',
+        },
+        {
+          message: 'Missing JSDoc @param "root0" declaration.',
```

Comment:
This message is confusing, what is `root0`?

Reply:
This is part of the feature added in #498 --that when a destructured parameter is also missing its root declaration, that by default, the word "root" will be used followed by a 0-based auto-incrementing number for each subsequent root (one can avoid the numeric additions with specific config, or add additional items to an options array so that until all the items in the array are exhausted, it will cycle through the names, e.g., `["root", "cfg", "config"]` will add `@param root` then `@param cfg`, then `@config0`, with the last item auto-incrementing). You could see our discussion about it in the PR for more background.

Reply:
Though I think it makes sense to use `root0` in the fixes, subject to user configuration (disablable by setting `enableRootFixer` to `false`), we could give a message like `"Missing JSDoc @param destructuring root object declaration"`. Does that sound better? And I'd be personally ok having `enableRootFixer` default to `false`, but sometimes I think it is good when the defaults are--within reason--a little over-aggressive since it informs users of the features and if they *don't* want it, they can come looking.

Reply:
Missing JSDoc @param declaration for destructured parameter `{bar}`.

Reply:
That could potentially become quite long though, no? e.g., `{bar, baz, foo: {def: {ghi}}, many, others}`

Reply:
I guess we could just slice it to a certain maximum length, adding the curly bracket always at the end.

Reply:
How should we differentiate in this test case though?
```js
      code: `
          /**
           *
           */
          function quux ({foo}) {

          }
      `,
      errors: [
        {
          message: 'Missing JSDoc @param "root0" declaration.',
        },
        {
          message: 'Missing JSDoc @param "root0.foo" declaration.',
        },
      ],
```

Report `Missing JSDoc @param "{foo}" declaration` twice?

Reply:
How should we differentiate in this test case though?
```js
      code: `
          /**
           *
           */
          function quux ({foo}) {

          }
      `,
      errors: [
        {
          message: 'Missing JSDoc @param "root0" declaration.',
        },
        {
          message: 'Missing JSDoc @param "root0.foo" declaration.',
        },
      ],
      output: `
          /**
           * @param root0
           * @param root0.foo
           */
          function quux ({foo}) {

          }
      `,
```

Report `Missing JSDoc @param "{foo}" declaration` twice?

Reply:
We don't want to report just once because the root object may or may not already be defined, and if not defined, there should be a separate message and fix.

Reply:
Btw, if it's ok, can we discuss this as a separate issue for fixing reporting messages? This discussion, though valid, is concerning a preexisting behavior (we refer to `root0` in other messages, not just in this PR). It would be nice to get this merged so the disabling functionality is available sooner.

Reply:
As far as the solution, I guess I can just change it to report:

> Missing JSDoc @param "{foo}" declaration
> Missing JSDoc @param "foo" in "{foo}" declaration